draft-ietf-tsvwg-sctp-dtls-chunk.md

```diff
@@ -1217,31 +1217,438 @@ some extension of the socket API.
+	uint64_t sdk_epoch;
+	uint8_t *sdk_conn_id;
+	uint8_t *sdk_key;
+	uint8_t *sdk_iv;
+	uint8_t *sdk_sn_key;
+};
+~~~
+
+``sdk_assoc_id``:
+: This parameter is ignored for one-to-one style sockets.
+  For one-to-many style sockets, this parameter indicates upon which
+  SCTP association the caller is performing the action.
+  It is an error to use ``SCTP_{FUTURE|CURRENT|ALL}_ASSOC``.
+
+``sdk_cipher_suit``:
+: The cipher suit for which the keys are used.
+
+``sdk_restart``:
+: If the value is ``0``, the regular keys are added, if a value different
+  from ``0`` is used, the restart keys are added.
+
+``sdk_conn_id_len``:
+: The length of the DTLS connection identifier specified in ``sdk_conn_id``.
+
+``sdk_key_len``:
+: The length of the initialization vector specified in ``sdk_key``.
+
+``sdk_iv_len``:
+: The length of the initialization vector specified in ``sdk_iv``.
+
+``sdk_sn_key_len``:
+: The length of the sequence number key specified in ``sdk_sn_key``.
+
+``sdk_unused``:
+: This field is ignored.
+
+``sdk_epoch``:
+: The epoch for which the keys are added.
+
+``sdk_conn_id``:
+: A pointer to the connection identifier for which the keys are added.
+  Using ``NULL`` means that no connection identifier is used.
+
+``sdk_key``:
+: A pointer to the key.
+
+``sdk_iv``:
+: A pointer to the initialization vector.
+
+``sdk_sn_key``:
+A pointer to the sequence number key.
+
+This socket option can only be used on SCTP endpoints in states other then
+``SCTP_LISTEN``, ``SCTP_COOKIE_WAIT`` and ``SCTP_COOKIE_ECHOED``.
+
+### Delete Receive Keys (``SCTP_DTLS_DEL_RECV_KEYS``)
+
+Using this socket option allows to remove a particular set of keys used for
+receiving DTLS chunks.
+
+The following structure is used as the ``option_value``:
+
+~~~ c
+struct sctp_dtls_keys_id {
+	sctp_assoc_t sdki_assoc_id;
+	uint8_t sdki_restart;
+	uint8_t sdki_conn_id_len;
+	uint16_t sdki_unused;
+	uint64_t sdki_epoch;
+	uint8_t *sdki_conn_id;
+}
+~~~
+
+``sdki_assoc_id``:
+: This parameter is ignored for one-to-one style sockets.
+  For one-to-many style sockets, this parameter indicates upon which
+  SCTP association the caller is performing the action.
+  It is an error to use ``SCTP_{FUTURE|CURRENT|ALL}_ASSOC``.
+
+``sdki_restart``:
+: If the value is ``0``, the regular keys are removed, if a value different
+  from ``0`` is used, the restart keys are removed.
+
+``sdki_conn_id_len``:
+: The length of the DTLS connection identifier specified in ``sdki_conn_id``.
+
+``sdki_unused``:
+: This field is ignored.
+
+``sdki_epoch``:
+: The epoch for which the keys are removed.
+
+``sdki_conn_id``:
+: A pointer to the connection identifier for which the keys are removed.
+  Using ``NULL`` means that no connection identifier is used.
+
+This socket option can only be used on SCTP endpoints in states other then
+``SCTP_CLOSED``, ``SCTP_LISTEN``, ``SCTP_COOKIE_WAIT`` and
+``SCTP_COOKIE_ECHOED``.
+
+### Set or Get Protection Enforcement (``SCTP_DTLS_ENFORCE_PROTECTION``)
```

Comment:
Shouldn't protection enforcement be per direction? I can see that it is possible to wait to set it until one have the keys in place in both directions, however I do wonder if it introduces issues with key-management that might intended to set it one direction and expects to be able to rely on the SCTP stack to only give protected packets in that direction. Realize this might be an issue in the abstract API. So lets land this and I file a seperate issue for it.

Reply:
See issue #79

Reply:
In my mind model, protection enforcement is only in the receive direction. I made the text clearer. In the send direction, this is controller by setting a key for sending. Once it a key is set, it is used.